agent/agentbootstrap/bootstrap.go

```diff
@@ -257,6 +257,20 @@ func (b *AgentBootstrap) Initialize(ctx context.Context) (_ *state.Controller, r
 		return nil, fmt.Errorf("%w %q during bootstrap", modelerrors.AgentVersionNotSupported, agentVersion)
 	}
 
+	// localModelRecordOP defines the bootstrap operation that should be run
+	// to establish the local model record in the controller model's database.
+	// We have two variants of this to handle the case when the user as set a
+	// custom agent stream to use for the controller model.
+	localModelRecordOp := modelbootstrap.CreateLocalModelRecord(
+		controllerModelUUID, controllerUUID, agentVersion,
+	)
+	if stateParams.ControllerModelConfig.AgentStream() != "" {
```

Comment:
Do we delete the agent stream from the controller model config before saving the config? If not we should.

Reply:
Yeah it is a good question. I had to look as I was worried I had forgot it. But the answer is yes we do. If you look in the bootstrap operation for model config we explicitly get rid of agent version and agent key.

I opted to make them a concern of the bootstrap operations not the agent bootstrap code so the agent bootstrap code doesn't get too cluttered with concerns.